AppCenterData/AppCenterData/MSData.m

```diff
@@ -403,9 +403,18 @@ - (void)listDocumentsWithType:(Class)documentType
       dataError = [self generateDisabledError:@"list" documentId:nil];
     } else if (![MSDocumentUtils isSerializableDocument:documentType]) {
       dataError = [self generateInvalidClassError];
+    } else if ([self.reachability currentReachabilityStatus] == NotReachable) {
```

Comment:
Doesn't it also prevent fetching the first page of data when we are offline (i.e. from the cache)?

Reply:
I think this logic would be fine only if we are testing for `continuationToken` being not `nil` at the same time. And we are probably missing some unit tests if that did not break anything.

Reply:
makes sense, updated the code, if continuation token is provided and device if offline, then it returns error. 
otherwise, it return the documents (including the first page)